app/jobs/upload_job.rb

```diff
@@ -86,30 +86,55 @@ def load_projects_data(locale, repository, owner)
   end
 
   def format_project(project_dir, locale, repository, owner)
-    components = []
-    images = []
-    proj_config = {}
-
     data = project_dir.object
-    raise InvalidDirectoryStructureError, "The directory structure is incorrect and the job can't be processed." unless data.respond_to?(:entries)
+    validate_directory_structure(data)
 
-    data.entries.each do |file|
-      if file.name == 'project_config.yml'
-        proj_config = YAML.safe_load(file.object.text, symbolize_names: true)
+    proj_config_file = data.entries.find { |file| file.name == 'project_config.yml' }
+    proj_config = YAML.safe_load(proj_config_file.object.text, symbolize_names: true)
 
-        # Skip if build is set to false (for backwards compatibility the build must happen if the key is not present)
-        if proj_config[:build] == false
-          @skip_job = true
-          break
-        end
-      elsif file.object.text
-        components << component(file)
+    if proj_config[:build] == false
+      @skip_job = true
+      return proj_config
+    end
+
+    files = data.entries.reject { |file| file.name == 'project_config.yml' }
+    categorized_files = categorize_files(files, project_dir, locale, repository, owner)
+
+    { **proj_config, locale:, **categorized_files }
+  end
+
+  def validate_directory_structure(data)
+    raise InvalidDirectoryStructureError, 'The directory structure is incorrect and the job can\'t be processed.' unless data.respond_to?(:entries)
+  end
+
+  def categorize_files(files, project_dir, locale, repository, owner)
+    categories = {
+      components: [],
+      images: [],
+      videos: [],
+      audio: []
+    }
+
+    files.each do |file|
+      next if file.name == 'project_config.yml'
```

Comment:
is this necessary since you're calling reject for the config above?

Reply:
Ah I'd missed this, thanks! 🗑️